### PR TITLE
nrf_modem: doc: sockets: add info on CID and CONN_SAVE socket options

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Expand the CID, CID_STATUS and CONN_SAVE socket option documentation.

### CODE REVIEW

**Review by lemrey — APPROVED** [review]
Thank you

File: nrf_modem/doc/sockets.rst [hunk 1]
```diff
@@ -275,17 +275,19 @@ NRF_SO_SEC_HANDSHAKE_STATUS
...
    The socket option is supported from modem firmware v1.3.x, where x is greater than or equal to 5, and v2.x.x.
    The following values are accepted:
 
-   * :c:macro:`NRF_SO_SEC_DTLS_CID_DISABLED` - The DTLS connection ID is not used, and the connection ID extension is not included in the client hello.
-   * :c:macro:`NRF_SO_SEC_DTLS_CID_SUPPORTED` - The DTLS connection ID is supported but not used, and the connection ID extension with a zero-length CID is included in the client hello.
-   * :c:macro:`NRF_SO_SEC_DTLS_CID_ENABLED` - The DTLS connection ID is used, and the connection ID extension with a valid CID is included in the client hello.
+   * :c:macro:`NRF_SO_SEC_DTLS_CID_DISABLED` - The connection ID extension is not included in the client hello. Hence, the DTLS connection ID is not used.
+   * :c:macro:`NRF_SO_SEC_DTLS_CID_SUPPORTED` - The connection ID extension with a zero-length CID is included in the client hello. Hence, the modem will accept a DTLS connection ID from the server.
+   * :c:macro:`NRF_SO_SEC_DTLS_CID_ENABLED` - The connection ID extension with a valid CID is included in the client hello. Hence, DTLS connection ID support is requested by the modem.
```
Comment (wiba-nordic) [inline nrf_modem/doc/sockets.rst:284]:
```suggestion
   * :c:macro:`NRF_SO_SEC_DTLS_CID_DISABLED` - The connection ID extension is not included in the client hello, so the DTLS connection ID is not used.
   * :c:macro:`NRF_SO_SEC_DTLS_CID_SUPPORTED` - The connection ID extension with a zero-length CID is included in the client hello, so the modem will accept a DTLS connection ID from the server.
   * :c:macro:`NRF_SO_SEC_DTLS_CID_ENABLED` - The connection ID extension with a valid CID is included in the client hello, so the modem will request DTLS connection ID support.
```

File: nrf_modem/doc/sockets.rst [hunk 2]
```diff
@@ -275,17 +275,19 @@ NRF_SO_SEC_HANDSHAKE_STATUS
...
 
    The default is disabled.
 
 NRF_SO_SEC_DTLS_CID_STATUS
    Get the status of DTLS connection ID.
+   This socket option tells whether the connection ID is used in the current connection.
```
Comment (wiba-nordic) [inline nrf_modem/doc/sockets.rst:290]: At the moment, these lines seem to say the same thing. Is it still correct if Line 290 reads "The status tells whether..."?

File: nrf_modem/doc/sockets.rst [hunk 3]
```diff
@@ -298,6 +300,9 @@ NRF_SO_SEC_DTLS_CID_STATUS
 
 NRF_SO_SEC_DTLS_CONN_SAVE
    Save DTLS connection.
+   This socket option can be used to pause a session that is not frequently used by the application. Saving the session will free memory in the modem that can be used for other connections.
```
Comment (wiba-nordic) [inline nrf_modem/doc/sockets.rst:303]:
```suggestion
   This socket option can be used to pause a session that is not frequently used by the application.
   Saving the session will free memory in the modem, so the memory can be used for other connections.
```
I think it's supposed to say that the memory is what can be used for other connections. If it's the modem that can be used for other connections, apply only the line break.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Comment by eivindj-nordic** [comment]
@wiba-nordic I've addressed your comments. Please have another look :)

**Comment by wiba-nordic** [comment]
Looks good, I'll wait for doc build to finish before approving.

**Review by wiba-nordic — APPROVED** [review]
Nevermind, we're on nrfxlib not ncs.

File: CMakeLists.txt [hunk 4]
```diff
@@ -5,6 +5,8 @@
 #
 include(common.cmake)
 
+message("HELLO FROM NRFXLIB")
```
Comment (rlubos) [inline CMakeLists.txt:8]: Is that change really intended?

File: zephyr/module.yml [hunk 5]
```diff
@@ -3,3 +3,4 @@ build:
   kconfig: Kconfig.nrfxlib
   depends:
   - hal_nordic
+  cmake: .
```
Comment (rlubos) [inline zephyr/module.yml:6]: Ditto.

**Review by rlubos — CHANGES_REQUESTED** [review]
Please open a manifest PR (nrfxlib CI does not build documentation).

File: CMakeLists.txt (hunk 4 above)
Comment (eivindj-nordic) [reply CMakeLists.txt]: No, these should not be there :/ Updated it now!

**Review by rlubos — APPROVED** [review]
